### PR TITLE
Budget App Rails

### PR DESCRIPTION
**Pull request by jugosack**
### In this pull request are completed the following activities for the Microverse budget-app project:

**Features**

- Set up the repository and tools.
- Create your models according to the ERD diagram.
- Add validations and tests.
- Implement the authentication and authorization.
- Create the page to add a category.
- Create the home page.
- Create the page to add an expense.
- Create the expenses page.
- Make sure tests are still passing.
- Deploy the project
- Record a video of the project.
- Add README and a pull request for review.

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @jugosack 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 The PR has linters installed and they pass the check ✔️ 
👍🏽 The Readme file follows the microverse's template
👍🏽 The PR is professional with a good commit history and a description ✔️ 
👍🏽 The video recorded is really descriptive and professional ✔️ 
👍🏽 The app works almost as it should be working ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [ ] The video contains a highlight about a piece of the code the student finds interesting or is proud of making.
- [ ] While creating a transaction, the user fills out a form to create a new transaction with name (mandatory), amount (mandatory), and categories (mandatory at least one) => When I click on create without filling the form, I got a 500 internal server error, would you address this one and return a validation message instead of crashing the server?

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] On Creating a category, the user fills out a form to create a new category, indicating their name and icon (both mandatory) => If I click on create without filling out the form, it does not show any error message, I would recommend you to show validation message
- [ ] Would recommend you install the github action for test cases so that you have them up and running on github action on every push

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jugosack** [comment]
@descholar-ceo Thanks for your time and effort. I have taken into consideration your remarks and the codebase is fixed.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,146 @@
...
-* Ruby version
+# Budget App <a name="about-project"></a>
 
-* System dependencies
+> This mobile web app allows users to create categories and add transactions to each category. The app will show the total amount spent on each category. The app is built with Ruby on Rails. The app is deployed on Render.
 
-* Configuration
+## video demo 🎥
+
+[Video demo here](https://www.loom.com/share/edb4f04d81e64e20b2953e79e519982d)
```
Comment (tugsem) [inline README.md:31]: - [ ] As the previous reviewer required, kindly record a new presenting video and don't forget to talk about a highlight about a piece of the code you find interesting or are proud of making. 🙏

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hello @jugosack 👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Highlights 🚀
✔️Good job on resolving the issue on the transactions' form

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (jugosack) [reply README.md]: @tugsem Thanks for your time and effort. I have taken into consideration your remarks and here is the link for the newly recorded video https://www.loom.com/share/1b2427fb867c4e1c8d2419d57991c1f4.

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @jugosack    , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  a. 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] No linter errors
- [x] Nice UI design 💯
- [x] Tests are passing 😎 
 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**